gooddata-pandas/gooddata_pandas/dataframe.py

```diff
@@ -346,6 +344,42 @@ def for_exec_result_id(
         if result_cache_metadata is None:
             result_cache_metadata = self.result_cache_metadata_for_exec_result_id(result_id=result_id)
 
+        df, _ = convert_execution_response_to_dataframe(
+            execution_response=BareExecutionResponse(
+                actions_api=self._sdk.client.actions_api,
+                workspace_id=self._workspace_id,
+                execution_response=models.AfmExecutionResponse(
+                    result_cache_metadata.execution_response, _check_type=False
+                ),
+            ),
+            result_cache_metadata=result_cache_metadata,
+            label_overrides=label_overrides,
+            result_size_dimensions_limits=result_size_dimensions_limits,
+            result_size_bytes_limit=result_size_bytes_limit,
+            use_local_ids_in_headers=use_local_ids_in_headers,
+        )
+        return df
+
+    def for_exec_result_id_with_metadata(
+        self,
+        result_id: str,
+        label_overrides: Optional[LabelOverrides] = None,
+        result_cache_metadata: Optional[ResultCacheMetadata] = None,
+        result_size_dimensions_limits: ResultSizeDimensions = (),
+        result_size_bytes_limit: Optional[int] = None,
+        use_local_ids_in_headers: bool = False,
+    ) -> Tuple[pandas.DataFrame, Optional[DataFrameMetadata]]:
```

Comment:
Do we really need an extra method for including metadata? Can we just extend for_exec_result_id with `include_metadata: bool = False` parameter?

Reply:
That's a good question. If you agree with introducing breaking change, then I can just enhance the existing one.
IMO it's fine as I don't expect anybody is using it on prod today.

Reply:
I do not think that it is a breaking change because the behavior of `for_exec_result_id` will not be affected and the parameter `include_metadata` will be optional (having default value set to `False`).

Reply:
fixed